File: algo-contest-1-17-07-2022/A/solution.py
queries = int(input())

while (queries):
    queries -= 1
    s = input()
    n = len(s)
    ans = 0
    f = dict()
    
    for char in s:
        f[char] = 1
        if len(f) > 3:
            f.clear()
            f[char] = 1
            ans += 1
    
    if len(f) > 0:
        ans += 1
    
    print(ans)

File: algo-contest-1-17-07-2022/A/solution.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int queries;
    cin >> queries;
    while (queries--) {
        string s;
        cin >> s;
        int n = s.length(), ans = 0;

        set < char > f;
        for (char chr: s) {
            f.insert(chr);
            if ((int)f.size() > 3) {
                ans++;
                f.clear();
                f.insert(chr);
            }
        }
        if (f.empty() == false)
            ans++;
        cout << ans << "\n";
    }
    return 0;
}